src/main/java/no/ndla/taxonomy/service/NodeService.java

```diff
@@ -117,15 +117,19 @@ public List<NodeDTO> getNodesByType(
             Optional<Boolean> includeContexts,
             boolean filterProgrammes) {
         final List<NodeDTO> listToReturn = new ArrayList<>();
-        var ids = nodeRepository.findIdsFiltered(
-                nodeType,
-                metadataFilters.getVisible(),
-                metadataFilters.getKey(),
-                metadataFilters.getLikeQueryValue(),
-                contentUri,
-                contextId,
-                isRoot,
-                isContext);
+        List<Integer> ids;
+        if (contextId.isPresent()) {
+            ids = nodeRepository.findIdsByContextId(contextId);
```

Comment:
Det her betyr vel gjerne at de andre filterene ikke fungerer når man spør om contextId?
Bør vi ha et eget endepunkt eller tåler vi at oppførselen blir litt uforventet om man ikke vet hva man driver med? :smile:

Reply:
Eg har ombestmt meg. Ideellt sett skal contextId være unikt slik at du aldri vil få to noder med samme contextId, så om du filterer på dette så er det kun en som kan returneres.